packages/neutron-sdk/src/bindings/query.rs

```diff
@@ -55,18 +60,327 @@ pub enum NeutronQuery {
+#[serde(rename_all = "snake_case")]
+pub enum Liquidity {
+    PoolReserves(PoolReserves),
+    LimitOrderTranche(LimitOrderTranche),
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct PoolReserves {
+    key: PoolReservesKey,
+    reserves_maker_denom: Int128,
+    price_taker_to_maker: PrecDec,
+    price_opposite_taker_to_maker: PrecDec,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct PoolReservesKey {
+    trade_pair_id: TradePairID,
+    tick_index_taker_to_maker: i64,
+    fee: Option<u64>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct InactiveLimitOrderTrancheResponse {
+    inactive_limit_order_tranche: LimitOrderTranche,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct AllInactiveLimitOrderTrancheResponse {
+    inactive_limit_order_tranche: Vec<LimitOrderTranche>,
+    pagination: Option<PageResponse>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct AllPoolReservesResponse {
+    pool_reserves: Vec<PoolReserves>,
+    pagination: Option<PageResponse>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct PoolReservesResponse {
+    pool_reserves: PoolReserves,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct EstimateMultiHopSwapResponse {
+    coin_out: Coin,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct EstimatePlaceLimitOrderResponse {
+    // Total amount of coin used for the limit order
+    // You can derive makerLimitInCoin using the equation: totalInCoin = swapInCoin + makerLimitInCoin
+    total_in_coin: Coin,
+    // Total amount of the token in that was immediately swapped for swapOutCoin
+    swap_in_coin: Coin,
+    // Total amount of coin received from the taker portion of the limit order
+    // This is the amount of coin immediately available in the users account after executing the
+    // limit order. It does not include any future proceeds from the maker portion which will have withdrawn in the future
+    swap_out_coin: Coin,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct PoolResponse {
+    pool: Pool,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct Pool {
+    #[serde(default)]
+    id: u64,
+    lower_tick0: Option<PoolReserves>,
+    lower_tick1: Option<PoolReserves>,
```

Comment:
isn't it upper_tick1?
https://github.com/neutron-org/neutron/blob/77b35712d6fbd391f289828df422a886775b7088/proto/neutron/dex/pool.proto#L15